examples/thermoElastic1Dhessian/model.i

```diff
@@ -0,0 +1,195 @@
+[Mesh]
+  [gmg]
+    type = GeneratedMeshGenerator
+    dim = 1
+    nx = 5
+    xmin = 0
+    xmax = 2
+  []
+[]
+
+[Variables]
+  [ux]
+  []
+  [T]
+  []
+[]
+
+[AuxVariables]
+  [dummy]
+  []
+[]
+
+[Kernels]
+  [heat_conduction]
+    type = MatDiffusion
+    variable = T
+    diffusivity = 1.0
+  []
+  [div_sigma_x]
+    type = StressDivergenceTensors
+    variable = ux
+    displacements = 'ux'
+    component = 0
+    volumetric_locking_correction = false
+  []
+[]
+
+[BCs]
+  [left_ux]
+    type = DirichletBC
+    variable = ux
+    boundary = left
+    value = 0.0
+  []
+  [left_T]
+    type = DirichletBC
+    variable = T
+    boundary = left
+    value = 0.0
+  []
+  [right_T]
+    type = FunctionNeumannBC
+    variable = T
+    boundary = right
+    function = Q
+  []
+[]
+
+[Materials]
+  [thermal_strain]
+    type = ComputeThermalExpansionEigenstrain
+    temperature = T
+    thermal_expansion_coeff = '0.001'
+    stress_free_temperature = 0.0
+    eigenstrain_name = thermal_strain
+  []
+  [alpha_material]
+    type = GenericFunctionMaterial
+    prop_names = 'alpha_material'
+    prop_values = alpha
+  []
+  [strain]
+    type = ComputeSmallStrain
+    displacements = 'ux'
+    eigenstrain_names = thermal_strain
+  []
+  [stress]
+    type = ComputeLinearElasticStress
+  []
+  [elasticity_tensor]
+     type = ComputeVariableIsotropicElasticityTensor
+     args = dummy
```

Comment:
args is now coupled_variable and it is telling the material that youngs modulus and poissons ratio are a function of that variable so that it can be included in the off diagonal jacobian.  It is usually temperature for temperature dependent properites.  For this case I don't think you need dummy.